python/pyspark/sql/connect/dataframe.py

```diff
@@ -211,14 +212,66 @@ def filter(self, condition: Expression) -> "DataFrame":
             plan.Filter(child=self._plan, filter=condition), session=self._session
         )
 
-    def first(self) -> Optional["pandas.DataFrame"]:
-        return self.head(1)
+    def first(self) -> Optional[Row]:
+        """Returns the first row as a :class:`Row`.
+
+        .. versionadded:: 3.4.0
+
+        Returns
+        -------
+        :class:`Row`
+           First row if :class:`DataFrame` is not empty, otherwise ``None``.
+        """
+        return self.head()
```

Comment:
maybe this is copied from PySpark, but isn't it better to signal intent here with a 1 as explicit param?

Reply:
This is implementation details now but updated.

Reply:
ah actually we cannot. `self.head()` returns `Optional[Row]` but self.head returns `List[Row]`. `self.head()` is to make sure mypy check pass.